fastly/secret_store.go

```diff
@@ -379,3 +388,91 @@ func (c *Client) DeleteSecret(i *DeleteSecretInput) error {
 	}
 	return resp.Body.Close()
 }
+
+// ClientKey is an X25519 public key that can be used with
+// golang.org/x/crypto/nacl/box to encrypt secrets locally before
+// uploading them to the Fastly API.  A client key is valid only for a
+// short amount of time, and should be used immediately.  The key is not
+// valid after the ExpiresAt time.
+//
+// Client keys are signed, and the attached signature must be validated
+// using the public signing key before it is used.  A ValidateSignature
+// method is provided for this purpose.
+type ClientKey struct {
+	PublicKey []byte    `json:"public_key"`
+	Signature []byte    `json:"signature"`
+	ExpiresAt time.Time `json:"expires_at"`
+}
+
+// ValidateSignature reports if the signingKey was used to generate the
+// client key's signature.  It must be a valid Ed25519 public key, and
+// it will panic if len(signingKey) is not ed25519.PublicKeySize.
+// https://pkg.go.dev/crypto/ed25519#PublicKeySize
+func (ck *ClientKey) ValidateSignature(signingKey ed25519.PublicKey) bool {
+	return ed25519.Verify(signingKey, ck.PublicKey, ck.Signature)
+}
+
+// Encrypt uses the client key to encrypt the provided plaintext
+// using a libsodium-compatible sealed box.
+// https://pkg.go.dev/golang.org/x/crypto/nacl/box#SealAnonymous
+// https://libsodium.gitbook.io/doc/public-key_cryptography/sealed_boxes
+func (ck *ClientKey) Encrypt(plaintext []byte) ([]byte, error) {
+	if len(ck.PublicKey) != 32 {
```

Comment:
Should we use `ed25519.PublicKeySize` instead of the 'magic number'?
```suggestion
	if len(ck.PublicKey) != ed25519.PublicKeySize {
```

Reply:
No, because these are actually not the same thing.  The _signing key_ is an Ed25519 public key, and uses the APIs from https://pkg.go.dev/crypto/ed25519.  The _client key_ is an X25519 public key, and uses the APIs from https://pkg.go.dev/golang.org/x/crypto/nacl/box.  They're derived from similar things but aren't quite the same.  [This Stack Overflow answer](https://crypto.stackexchange.com/a/68129) goes into some details on the differences.

Both are 32 bytes long so it would work, but it doesn't feel quite right to use the `ed25519` values here.  Unfortunately, the `nacl/box` package hardcodes 32-byte arrays throughout its API and doesn't define a constant for this.